addons/binding/org.openhab.binding.xiaomivacuum/src/main/java/org/openhab/binding/xiaomivacuum/internal/RoboCommunication.java

```diff
@@ -0,0 +1,113 @@
+/**
+ * Copyright (c) 2010-2017 by the respective copyright holders.
+ *
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ */
+
+package org.openhab.binding.xiaomivacuum.internal;
+
+import java.net.DatagramPacket;
+import java.net.DatagramSocket;
+import java.net.InetAddress;
+
+import org.openhab.binding.xiaomivacuum.XiaomiVacuumBindingConstants;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * The {@link RoboCommunication} is responsible for communications with the vacuum
+ *
+ * @author Marcel Verpaalen - Initial contribution
+ */
+public class RoboCommunication {
+
+    private static final int PORT = 54321;
+    private static final int TIMEOUT = 5000;
+
+    private final static Logger logger = LoggerFactory.getLogger(RoboCommunication.class);
+
+    private String ip;
+    private byte[] token;
+    private byte[] serial;
+    private int id;
+
+    public RoboCommunication(String ip, byte[] token) throws Exception {
```

Comment:
doing remote communication and throwing Exception is a little surprising for a constrictor. How about spending a separate `initialize()`  method for it?